log4j-core-test/src/test/java/org/apache/logging/log4j/core/filter/HttpThreadContextMapFilterTest.java

```diff
@@ -16,183 +16,98 @@
+import static com.github.tomakehurst.wiremock.client.WireMock.ok;
+import static org.assertj.core.api.Assertions.assertThat;
 
-import java.io.File;
-import java.io.IOException;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.nio.file.StandardCopyOption;
-import java.util.Base64;
-import java.util.Enumeration;
-import java.util.concurrent.CountDownLatch;
-import java.util.concurrent.TimeUnit;
-import javax.servlet.ServletException;
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
-import org.apache.logging.log4j.Logger;
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.common.FileSource;
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration;
+import com.github.tomakehurst.wiremock.extension.Parameters;
+import com.github.tomakehurst.wiremock.extension.ResponseTransformer;
+import com.github.tomakehurst.wiremock.http.Request;
+import com.github.tomakehurst.wiremock.http.Response;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.concurrent.atomic.AtomicInteger;
 import org.apache.logging.log4j.ThreadContext;
-import org.apache.logging.log4j.core.Appender;
 import org.apache.logging.log4j.core.LoggerContext;
-import org.apache.logging.log4j.core.config.Configurator;
 import org.apache.logging.log4j.core.test.appender.ListAppender;
-import org.eclipse.jetty.http.HttpHeader;
-import org.eclipse.jetty.server.Server;
-import org.eclipse.jetty.server.ServerConnector;
-import org.eclipse.jetty.servlet.DefaultServlet;
-import org.eclipse.jetty.servlet.ServletContextHandler;
-import org.eclipse.jetty.servlet.ServletHolder;
-import org.junit.Assert;
-import org.junit.jupiter.api.AfterAll;
-import org.junit.jupiter.api.AfterEach;
-import org.junit.jupiter.api.Assertions;
-import org.junit.jupiter.api.BeforeAll;
+import org.apache.logging.log4j.core.test.junit.LoggerContextSource;
+import org.apache.logging.log4j.core.test.junit.Named;
+import org.apache.logging.log4j.spi.ExtendedLogger;
 import org.junit.jupiter.api.Test;
 
 /**
- * Unit test for simple App.
+ * Tests {@link ThreadContextMapFilter} using a WireMock stub.
  */
-public class HttpThreadContextMapFilterTest implements MutableThreadContextMapFilter.FilterConfigUpdateListener {
+public class HttpThreadContextMapFilterTest {
 
-    private static final String BASIC = "Basic ";
-    private static final String expectedCreds = "log4j:log4j";
-    private static Server server;
-    private static final Base64.Decoder decoder = Base64.getDecoder();
-    private static int port;
-    static final String CONFIG = "log4j2-mutableFilter.xml";
-    static LoggerContext loggerContext = null;
-    static final File targetFile = new File("target/test-classes/testConfig.json");
-    static final Path target = targetFile.toPath();
-    CountDownLatch updated = new CountDownLatch(1);
+    @Test
+    @LoggerContextSource("HttpThreadContextMapFilterTest.xml")
```

Comment:
A couple of months ago I introduced a default value for a logger context configuration: if you move the config file to `src/test/resources/org/apache/logging/log4j/core/filter/HttpThreadContextMapFilterTest.xml` (i.e. same package and name as the test), you don't need to provide a value to the `@LoggerContextSource` annotation.

Reply:
Fixed in 4838ceba9a098b78c51d7fe16bcd4e208fda85c6.